IGNITE-13796 docs: update docs and examples for kubernetes module (#8532)
apache/ignite: docs/_docs/thin-clients/java-thin-client.adoc

include::{sourceCodeFile}[tag=partition-awareness,indent=0]
----

> If list of server nodes is dynamically changing or scaling than it is possible to configure connection with custom implementation of `ClientAddressFinder`. It should provide actual server addresses collection for every time client ask it.

Reviewer: ```suggestion
If a list of server nodes is dynamically changing or scaling, then it is possible to configure the connection with custom implementation of `ClientAddressFinder`. It should provide a number of current server addresses every time a client asks for them.
```

The following code sample illustrates how to use it.

[source, java]
----
include::{sourceCodeFile}[tag=client-address-finder,indent=0]
----

> Currently Ignite provides the only implementation of this interface. `ThinClientKubernetesAddressFinder` is created to handle scalable Kubernetes environment.

Reviewer: ```suggestion
Currently, Ignite provides the only implementation of this interface. `ThinClientKubernetesAddressFinder` is created to handle scalable Kubernetes environment.
```

Reviewer 2: I don't understand what the sentence means. You say "Ignite provides the only implementation of this interface". What interface is assumed? Why is it important to mention this only implementation?

Instead, provide a bullet-point list explaining the code snippet. The snippet has the `fetchServerAddresses` method that is not implemented. It seems that you are trying to say that "The fetchServerAddress() method shows how an example implementation might look like if you want clients to retrieve the server nodes' addresses dynamically. If you are running in a Kubernetes environment then use the `ThinClientKubernetesAddressFinder` in your thin client configuration to enable the automatic resolution of the server addresses on the client side".  Nikita please help to adjust the latter if Maxim confirms that my understanding is correct.

Author: @dmagda @Nikita-tech-writer  
To mention this only implementation is may be important to provide an existing example of the interface implementation.

Yes, your understanding is correct.

Author: @Nikita-tech-writer I've updated the docs. Could you please check it?

Reviewer: @timoninmaxim Everything looks good to me! I approve the PR.
